test/parallel_api/iterator/permutation_iterator_parallel_find.pass.cpp

```diff
@@ -80,8 +82,10 @@ run_algo_tests()
 #if TEST_DPCPP_BACKEND_PRESENT
     // Run tests on <USM::shared, USM::device, sycl::buffer> + <all_hetero_policies>
     // dpl::find, dpl::find_if, dpl::find_if_not -> __parallel_find -> _parallel_find_or
-    test1buffer<sycl::usm::alloc::shared, ValueType, test_find<ValueType, PermItIndexTag>>();
-    test1buffer<sycl::usm::alloc::device, ValueType, test_find<ValueType, PermItIndexTag>>();
+    test1buffer<sycl::usm::alloc::shared, ValueType, test_find<ValueType, PermItIndexTag, new_kernel_name<KernelName, 10>>,
+                                                     test_find<ValueType, PermItIndexTag, new_kernel_name<KernelName, 15>>>();
+    test1buffer<sycl::usm::alloc::device, ValueType, test_find<ValueType, PermItIndexTag, new_kernel_name<KernelName, 20>>,
+                                                     test_find<ValueType, PermItIndexTag, new_kernel_name<KernelName, 25>>>();
```

Comment:
Is there anything which requires the numbers used here `10`, `15`, `20`, `25`, or are they arbitrary (and just have to be unique from each other)?

Also, similar to my other comment, I wonder if these unique names could be handled instead with wrapper layers at each "branch" for an individual test.  One wrapper layer here, one wrapper layer inside of `test1buffer` to split between usm and buffers.

Reply:
They just have to be unique from each other.

Reply:
Let's discuss about of `permutation_iterator_parallel_transform_scan.pass.cpp` test file and the test `test_remove_if`.
If I will call test in this file as
`test1buffer<sycl::usm::alloc::shared, ValueType, test_remove_if<ValueType, PermItIndexTag, KernelName>>();`
I still have a lot of compile errors when unnamed lambda is switched off.

Why these errors happens?

The happens because historically inside [`TestUtils::test1buffer`] function we have two calls whech using Krenel names inside:
1) ```C++
            invoke_on_all_hetero_policies<0>()(create_test_obj<TestValueType, TestName>(test_base_data),
                                               inout1_offset_first, inout1_offset_first + n,
                                               n);
```
2) ```C++
            invoke_on_all_hetero_policies<1>()(create_test_obj<TestValueType, TestName>(test_base_data),
                                               inout1_offset_first, inout1_offset_first + n,
                                               n);
```
As confirmation of that it we comment (2) there is no errors anymore.
But on this level I don't know how to explain to fully specialized class name `TestName` some additional template things. I believe it's  simple impossible.

That's is the main reason why we should generate some unique Kernel names for tests inside `main` functions.

Reply:
If you want you may take a look at this state in the separate temporary branch
https://github.com/oneapi-src/oneDPL/tree/dev/skopienko/fix_permutation_iterator_compile_error_tests_fix_review:
I reproduced here compile error for test `test_remove_if` from the file `permutation_iterator_parallel_transform_scan.pass.cpp`